src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -834,6 +1089,77 @@ private void AddMetaKeyValues(int i, DataViewSchema.Annotations.Builder builder)
             }
 
             protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer) => _parent.GetGetterCore(input, iinfo, out disposer);
+
+            private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, string srcVariable, string dstVariable)
+            {
+                string opType;
+                OnnxNode murmurNode;
+
+                opType = "MurmurHash3";
+                string murmurOutput = (_srcTypes[iinfo].RawType == typeof(ReadOnlyMemory<char>)) ? dstVariable: ctx.AddIntermediateVariable(_dstTypes[iinfo], "MurmurOutput");
+                var srcType = _srcTypes[iinfo].GetItemType().RawType;
+
+                // Numeric input types are limited to those supported by the Onnxruntime MurmurHash operator, which currently only supports
+                // uints and ints. Thus, ulongs, longs, doubles, floats, and booleans are not supported.
+                if (srcType == typeof(ushort) || srcType == typeof(short) || srcType == typeof(sbyte) || srcType == typeof(byte))
+                {
+                    var castOutput = ctx.AddIntermediateVariable(NumberDataViewType.UInt32, "CastOutput", true);
+                    var castNode = ctx.CreateNode("Cast", srcVariable, castOutput, ctx.GetNodeName(opType), "");
+                    castNode.AddAttribute("to", NumberDataViewType.UInt32.RawType);
+                    murmurNode = ctx.CreateNode(opType, castOutput, murmurOutput, ctx.GetNodeName(opType), "com.microsoft");
+                }
+                else if (srcType == typeof(uint) || srcType == typeof(int) || srcType == typeof(ReadOnlyMemory<char>))
+                {
+                    murmurNode = ctx.CreateNode(opType, srcVariable, murmurOutput, ctx.GetNodeName(opType), "com.microsoft");
+                }
+                else
+                {
+                    return false;
+                }
+
+                murmurNode.AddAttribute("positive", 1);
```

Comment:
also you have to note that there is no support for useOrderedHashing yet.

Reply:
I did support OrderHashing - take a look at  line 1143

Reply:
This takes care of the scalar column case, but not of vector columns.
(see how ordered hashing is computed in lines 1008-1020 in this file). I'm not sure how this can be handled in ONNX, does the `MurmurHash3` operator handle vectors at all (ordered or unordered)?

---
In reply to: [413203701](https://github.com/dotnet/machinelearning/pull/5013#discussion_r413203701) [](ancestors = 413203701)

Reply:
Agreed. I'll make a note about the lack of support for vectors.

Reply:
Are vectors not supported at all? Or did you mean just for the ordered case?
If vectors are supported, just not ordered, then the change you made to return `false` in the ordered case (line 1144), should be made only if the input is a vector - for scalars it can still do what you did before:

```
seed = Hashing.MurmurRound(seed, 0);
```
If you meant that vectors are not supported at all, then that would be the first thing you need to check, and return false in case of vector inputs.


---
In reply to: [414090190](https://github.com/dotnet/machinelearning/pull/5013#discussion_r414090190) [](ancestors = 414090190)

Reply:
yeah i meant only for vectors, 

---
In reply to: [415255169](https://github.com/dotnet/machinelearning/pull/5013#discussion_r415255169) [](ancestors = 415255169,414090190)